File: tests/transpiler/x/cs/closure.cs
// Mochi 0.10.31 - generated 2025-07-19 16:18:19 UTC
using System;

class Program {
    static Func<int, int> makeAdder(int n) {
    return (int x) => (x + n);
}
    static void Main() {
        var add10 = makeAdder(10);
        Console.WriteLine(add10(7));
    }
}

File: tests/transpiler/x/java/closure.java
public class Main {
    static java.util.function.IntUnaryOperator add10 = makeAdder(10);

    static java.util.function.IntUnaryOperator makeAdder(int n) {
        return (int x) -> x + n;
    }
    public static void main(String[] args) {
        System.out.println(add10.applyAsInt(7));
    }
}
